addons/survey/static/src/js/survey_form.js

```diff
@@ -1158,20 +1161,38 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
     * @private
     */
     _getInactiveConditionalQuestionIds: function () {
-        var self = this;
-        var inactiveQuestionIds = [];
-        if (this.options.triggeredQuestionsByAnswer) {
-            Object.keys(this.options.triggeredQuestionsByAnswer).forEach(function (answerId) {
-                if (!self.selectedAnswers.includes(parseInt(answerId))) {
-                     self.options.triggeredQuestionsByAnswer[answerId].forEach(function (questionId) {
-                        inactiveQuestionIds.push(questionId);
-                     });
-                }
-            });
+        const inactiveQuestionIds = [];
+        for (const [questionId, answerIds] of Object.entries(this.options.triggeringAnswersByQuestion || {})) {
+            if (!answerIds.some(answerId => this.selectedAnswers.includes(parseInt(answerId)))) {
+                inactiveQuestionIds.push(parseInt(questionId));
+            }
         }
         return inactiveQuestionIds;
     },
 
+    /**
+     * Apply visibility rules of conditional questions.
+     *
+     * @param {Number[] | String[] | Set | undefined} questionIds Conditional questions ids
+     */
+    _renderConditionalQuestionsVisibility: function(questionIds) {
```

Comment:
Actually this one should be "_toggleConditionalQuestionsVisibility" ^^

The other one could be "_checkConditionalQuestionsConfiguration" ?

Note: arg, naming is hard :D